samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/ItemRemoveListener.java

```diff
@@ -0,0 +1,145 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **********************************************************************************/
+
+
+
+package org.sakaiproject.tool.assessment.ui.listener.author;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.faces.event.AbortProcessingException;
+import javax.faces.event.ActionEvent;
+import javax.faces.event.ActionListener;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.sakaiproject.event.cover.EventTrackingService;
+import org.sakaiproject.tool.assessment.facade.AgentFacade;
+import org.sakaiproject.tool.assessment.facade.AssessmentFacade;
+import org.sakaiproject.tool.assessment.facade.ItemFacade;
+import org.sakaiproject.tool.assessment.facade.SectionFacade;
+import org.sakaiproject.tool.assessment.services.ItemService;
+import org.sakaiproject.tool.assessment.services.QuestionPoolService;
+import org.sakaiproject.tool.assessment.services.assessment.AssessmentService;
+import org.sakaiproject.tool.assessment.ui.bean.author.AssessmentBean;
+import org.sakaiproject.tool.assessment.ui.bean.author.ItemAuthorBean;
+import org.sakaiproject.tool.assessment.ui.bean.authz.AuthorizationBean;
+import org.sakaiproject.tool.assessment.ui.bean.questionpool.QuestionPoolBean;
+import org.sakaiproject.tool.assessment.ui.listener.util.ContextUtil;
+
+
+/**
+ * <p>Title: Samigo</p>
+ * <p>Description: Sakai Assessment Manager</p>
+ * <p>Organization: Sakai Project</p>
+ */
+
+public class ItemRemoveListener implements ActionListener
+{
+  private static Logger log = LoggerFactory.getLogger(ItemModifyListener.class);
+  //private String scalename;  // used for multiple choice Survey
+
+  /**
+   * Standard process action method.
+   * @param ae ActionEvent
+   * @throws AbortProcessingException
+   */
+  public void processAction(ActionEvent ae) throws AbortProcessingException
+  {		
+	  ItemAuthorBean item = (ItemAuthorBean) ContextUtil.lookupBean("itemauthor");
+	  ItemService delegate = new ItemService();
+	  if(item.getItemToDelete() == null) return;
+	  Long deleteId= item.getItemToDelete().getItemId();
+	  ItemFacade itemf = delegate.getItem(String.valueOf(deleteId));
+	  // save the currSection before itemf.setSection(null), used to reorder question sequences
+	  SectionFacade  currSection = (SectionFacade) itemf.getSection();
+	  Integer  currSeq = itemf.getSequence();
+
+	  QuestionPoolService qpdelegate = new QuestionPoolService();
+	  if (qpdelegate.getPoolIdsByItem(deleteId.toString()) ==  null || qpdelegate.getPoolIdsByItem(deleteId.toString()).isEmpty()){
```

Comment:
You're calling `deleteId.toString()` and `String.valueOf(deleteId)` several times in this method. You should extract this value into a variable and reference that, rather than performing the same parsing multiple times.